src/coreclr/jit/morph.cpp

```diff
@@ -5198,7 +5198,7 @@ GenTree* Compiler::fgMorphArrayIndex(GenTree* tree)
                                                                      asIndex->Arr()->AsStrCon()->gtSconCPX, &length);
             if ((cnsIndex < length) && (str != nullptr))
             {
-                GenTree* cnsCharNode = gtNewIconNode(str[cnsIndex], elemTyp);
+                GenTree* cnsCharNode = gtNewIconNode(str[cnsIndex], TYP_INT);
```

Comment:
Unrelated change: `"str"[0]` used to be folded into a CNS_INT with TYP_USHORT type - no idea how it didn't assert anywhere